src/Data/Aeson/Commit.hs

```diff
@@ -36,15 +49,15 @@ runCommit (Commit f) = runExceptT f >>= either (fmap absurd) pure
 (o .:> k) cont = commit (o .: k) (\v -> cont v <?> Key k)
 
 -- | Turn a 'Parser' into a 'Commit'.
---   Unlike 'liftParser', the parser's failure is recoverable.
+--   Unlike 'commitToParser', the parser's failure is recoverable.
 --
--- > fromParser p = commit p pure
-fromParser :: Parser a -> Commit a
-fromParser p = commit p pure
+-- > tryParser p = commit p pure
+tryParser :: Parser a -> Commit a
+tryParser p = commit p pure
 
 -- | Turn a 'Parser' into a 'Commit'.
---   Unlike 'fromParser', the parser's failure is _not_ recoverable, i.e. the commit always succeeds.
+--   Unlike 'tryParser', the parser's failure is _not_ recoverable, i.e. the parse is always committed.
 --
--- > liftParser p = commit (pure ()) (\() -> p) = Commit (lift p)
-liftParser :: Parser a -> Commit a
-liftParser p = commit (pure ()) (const p)
+-- > commitToParser p = commit (pure ()) (\() -> p) = Commit (lift p)
+commitToParser :: Parser a -> Commit a
+commitToParser p = commit (pure ()) (const p)
```

Comment:
Bikeshedding: I don't think this name is an improvement, the `commit` reads more as the noun from `Commit` than the verb. I liked `liftParser` because it emphasises that it's the side-effect way to turn a `Parser` into `Commit`

Reply:
Are you OK with, `tryParser` instead of `fromParser`. I found the two names a bit too similar and it was difficult to keep in mind which had what effect (similar to the issue with `fail/throwError` when using `MonadFail/MonadError` in various monads with multiple ways of failure.

Reply:
Yeah, I think `try` captures pretty well what's going on